### ISSUE TITLE
Learn from flutter hashValues – no recursion, more better

### ISSUE THREAD

**Issue by kevmoo** [QuiverDart/quiver_hashcode#7]
See https://github.com/flutter/engine/blob/52bbe398a5f1d77651e3f017fd5be6cf26afd705/lib/ui/hash_codes.dart

**Comment by kevmoo** [comment]
CC @davidmorgan

**Comment by davidmorgan** [comment]
Fun. I guess what built_value generates now, e.g.

https://github.com/google/built_value.dart/blob/master/example/lib/collections.g.dart#L63

is a tiny bit faster, no optional parameters or if statements -- but costs a little code size.

**Comment by kevmoo** [comment]
@davidmorgan When you know statically what the number of args is, sure.

**Comment by davidmorgan** [comment]
Yes. Just making the point that generation is always going to be better :) ... source_gen FTW.

Still makes sense for quiver, though.
